### PR TITLE
{geo}[foss/2021b] TELEMAC-MASCARET v8p3r1 w/ Python 3.9.6

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by moravveji** [comment]
that's a good news.
I tried to take a peek at the easyconfig file, but could not find the relevant folder under the listing for `T`. Could you please point me to the relevant easyconfig/easyblock files?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3107.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/3c5d817d8b67706badf2faa725e6dbdd for a full test report.

File: easybuild/easyconfigs/t/TELEMAC-MASCARET/TELEMAC-MASCARET-8p3r1-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+install_cmd += "sed -i 's/^fflags_gfo:/fflags_gfo: -fallow-invalid-boz/g' $HOMETEL/configs/systel.easybuild.cfg && "
+install_cmd += "export SYSTELCFG=$HOMETEL/configs/systel.easybuild.cfg && "
+install_cmd += "export USETELCFG=easybuild && "
+
+install_cmd += "export SCALAPACKHOME=$EBROOTSCALAPACK && "
+install_cmd += "export METISHOME=$EBROOTMETIS && "
+install_cmd += "export SCOTCHHOME=$EBROOTSCOTCH && "
+install_cmd += "export MUMPSHOME=$EBROOTMUMPS && "
+install_cmd += "export HDF5HOME=$EBROOTHDF5 && "
+install_cmd += "config.py && compile_telemac.py -j %(parallel)s"
```
Comment (jfgrimm) [inline easybuild/easyconfigs/t/TELEMAC-MASCARET/TELEMAC-MASCARET-8p3r1-foss-2021b.eb:62]: personal taste, but I'd consider something like this more readable (less duplication with all the `install_cmd +=`)
```suggestion
_sed_cmd = ' '.join([
    "sed -i -e 's/S9.gfortran/easybuild/g'"
    # strip out use of AED, GOTM, MED (optional dependencies)
    "-e 's/\(^[a-z]*_aed:\).*/\\1/g' -e 's/\(^[a-z]*_gotm:\).*/\\1/g' -e 's/\(^[a-z]*_med:\).*/\\1/g'",
    '-e "s/-O2/$F90FLAGS/g"',      # replace -O2 with desired compiler options
    '-e "s/ -lblas/ $LIBBLAS/g"',  # fix linker option for BLAS
    # downgrade Fortran compiler error to warning
    "-e 's/^fflags_gfo:/fflags_gfo: -fallow-invalid-boz/g'"
    "$HOMETEL/configs/systel.easybuild.cfg",
])

_pre_install_cmd = ' && '.join([
    "export HOMETEL=$PWD",
    # add Python scripts to $PATH and $PYTHONPATH
    "export PATH=$HOMETEL/scripts/python3/:$PATH",
    "export PYTHONPATH=$HOMETEL/scripts/python3:$PYTHONPATH",
    # force python to flush its output
    "export PYTHONUNBUFFERED=1",

    "cp $HOMETEL/configs/systel.edf.cfg $HOMETEL/configs/systel.easybuild.cfg",
    _sed_cmd,
    "export SYSTELCFG=$HOMETEL/configs/systel.easybuild.cfg",
    "export USETELCFG=easybuild",

    "export SCALAPACKHOME=$EBROOTSCALAPACK",
    "export METISHOME=$EBROOTMETIS",
    "export SCOTCHHOME=$EBROOTSCOTCH",
    "export MUMPSHOME=$EBROOTMUMPS",
    "export HDF5HOME=$EBROOTHDF5",
])

install_cmd = "%(_pre_install_cmd)s && config.py && compile_telemac.py -j %%(parallel)s"
```

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node050.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/f459b367dd7bee6f2861f4632bb0dd65 for a full test report.

File: easybuild/easyconfigs/t/TELEMAC-MASCARET/TELEMAC-MASCARET-8p3r1-foss-2021b.eb (hunk 1 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/t/TELEMAC-MASCARET/TELEMAC-MASCARET-8p3r1-foss-2021b.eb:62]: @moravveji you can see the full easyconfig in the "files" tab of this PR: https://github.com/easybuilders/easybuild-easyconfigs/pull/16274/files

Comment (jfgrimm) [reply easybuild/easyconfigs/t/TELEMAC-MASCARET/TELEMAC-MASCARET-8p3r1-foss-2021b.eb:62]: (raw file [here](https://raw.githubusercontent.com/easybuilders/easybuild-easyconfigs/da879364bc7f8c8db0050fca3a8a822ddac6ee0f/easybuild/easyconfigs/t/TELEMAC-MASCARET/TELEMAC-MASCARET-8p3r1-foss-2021b.eb))

Comment (boegel) [reply easybuild/easyconfigs/t/TELEMAC-MASCARET/TELEMAC-MASCARET-8p3r1-foss-2021b.eb]: fixed in 7b731cbcbd (tweaked your suggested change a tiny bit)

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3116.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/0d2d576c600fa8db3d776fa4142d8136 for a full test report.

**Review by jfgrimm — APPROVED** [review]
LGTM

**Comment by jfgrimm** [comment]
Going in, thanks @boegel!

**Pull request merged**